### PR TITLE
generate_db.js: Refactor the code

### PR DESCRIPTION
**Pull request by ravener**
I just felt like rewriting the script to be cleaner. I switched to full async/await and ES6 modules, used the fetch API (available since Node.js 18) for the requests and overall tried to make the code look more modern.

The only downside i can think of is it's no longer parallel, I kind of made it sequential so only one network request is being done at a time, however if this is important I can very easily add that back.

I'm open to any suggestions that can improve readability.

### CODE REVIEW

File: generate-db.js [hunk 1]
```diff
@@ -1,78 +1,63 @@
...
-const path = require("node:path");
-const zlib = require("node:zlib");
+import { readFile, writeFile } from 'node:fs/promises';
+import { gunzip } from 'node:zlib';
+import { join } from 'node:path';
+import { promisify } from 'node:util';
+
+const gunzipAsync = promisify(gunzip);
 
-// TODO(@thunder-coding): Do not hardcode list of known architectures.
```
Comment (thunder-coding) [inline generate-db.js:7]: Thanks for reducing one TODO off my TODO list, really appreciate it!

**Review by thunder-coding — CHANGES_REQUESTED** [review]
This looks way cleaner. I had a similiar patch lying around in my junk local repo, but couldn't get enough time to look around getting it here. 

Just dont remove that TODO, and its ready to merge. (Will try to test things before though)

File: generate-db.js (hunk 1 above)
Comment (ravener) [reply generate-db.js:7]: Done!

**Comment by thunder-coding** [comment]
> The only downside i can think of is it's no longer parallel, I kind of made it sequential so only one network request is being done at a time, however if this is important I can very easily add that back.

Sequential performs better when testing locally (since bandwidth is often limited), for CI parallel gives a few seconds of boost (but I dont think there's much need of overengineering things)

**Comment by ravener** [comment]
> > The only downside i can think of is it's no longer parallel, I kind of made it sequential so only one network request is being done at a time, however if this is important I can very easily add that back.
> 
> Sequential performs better when testing locally (since bandwidth is often limited), for CI parallel gives a few seconds of boost (but I dont think there's much need of overengineering things)

It's not so much overengineering, basically I extract the code into a function and then then execute multiple instances, wrapping them in a `Promise.all` which is as simple as `await Promise.all(repos.map(processRepo));` this way it doesn't wait for individual repos but run the function for all repos and then wait for all of them to finish.

**Comment by thunder-coding** [comment]
> It's not so much overengineering, basically I extract the code into a function and then then execute multiple instances, wrapping them in a Promise.all which is as simple as await Promise.all(repos.map(processRepo)); this way it doesn't wait for individual repos but run the function for all repos and then wait for all of them to finish

Then go ahead :)

**Review by thunder-coding — APPROVED** [review]
LGTM. Working great. Thanks for the rewrite.

**Pull request merged**